infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/environment/KubernetesEnvironment.java

```diff
@@ -127,20 +160,45 @@ protected KubernetesEnvironment(
     protected final Map<String, ConfigMap> configMaps = new HashMap<>();
     protected final Map<String, String> attributes = new HashMap<>();
 
-    protected Builder() {}
+    protected Builder() {
+      this.internalEnvironment = null;
+      setType(TYPE);
+    }
+
+    public Builder(InternalEnvironment internalEnvironment) {
+      this.internalEnvironment = internalEnvironment;
+      setType(TYPE);
```

Comment:
Either set type here or on initialization

Reply:
Here we set predefined value in `Builder`, a client is able to set a new value, and then hold type is set to the created environment. Can you elaborate what is an issue with that?

Reply:
I see it is set in the field initialization step and in all the constructors. Something seems redundant.

Reply:
I don't see type initialization in any of `KubernetesEnvironment`s contructors but I see it in `OpenShiftEnvironment`. 
It was decided to leave type initialization in contructors since `OpenShiftEnvironment` has public contructors.